apps/backend/src/auth/auth.service.ts

```diff
@@ -0,0 +1,161 @@
+import {
+  BadRequestException,
+  ConflictException,
+  Injectable,
+  InternalServerErrorException,
+} from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { bcrypt } from 'bcryptjs';
+import { ParishService } from '../parish/parish.service';
+import { LoginDataDto, ParishDataDto } from './dto/login.dto';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly JwtService: JwtService,
+    private readonly parishService: ParishService
+  ) {}
+
+  /**
+   * This function authenticate the parish user when login
+   * @param input
+   * @returns
+   */
+  async authenticateParish(input: LoginDataDto): Promise<ParishDataDto> {
+    const user = await this.validate(input, 'parish');
+
+    if (!user) {
+      throw new BadRequestException('Bad Request', {
+        cause: new Error(),
+        description: 'Wrong email or password.',
+      });
+    } else if (typeof user === 'string') {
+      throw new ConflictException(user, {
+        cause: new Error(),
+        description:
+          'Your account is already in use! Please logout before logging again.',
+      });
+    }
+
+    return this.signIn(user);
+  }
+  /**
+   * This function authenticate administrators user when login
+   * @param input
+   * @returns
+   */
+  async authenticateAdmin(input: LoginDataDto): Promise<ParishDataDto> {
+    const user = await this.validate(input, 'admin');
+
+    if (!user) {
+      throw new BadRequestException('Bad Request', {
+        cause: new Error(),
+        description: 'Wrong email or password.',
+      });
+    } else if (typeof user === 'string') {
```

Comment:
Why do you need this emphasis `typeof user === 'string'` ? Just as `else` statement will have been sufficient